packages/headless/src/features/generated-answer/generated-answer-state.ts

```diff
@@ -54,10 +58,23 @@ export interface GeneratedAnswerState {
    * A list of indexed fields to include in the citations returned with the generated answer.
    */
   fieldsToIncludeInCitations: string[];
+  /**
+   * Whether the current answer stream should be aborted.
+   */
+  shouldAbortStream: boolean;
```

Comment:
We have a few state flags here and I'm wondering if we should group them together into one attribute with multiple states.
What we have:
- isLoading
- isStreaming
- shouldStartStream
- shouldAbortStream
ex:
```ts
    streamStatus: 'LOADING' | 'STREAM_PENDING' | 'STREAMING' | 'ABORT' | 'INACTIVE';
```

More of a state machine type of setup.
Just throwing this out there for your consideration.
I think it cleans up the state and potentially makes things simpler as we can't have weird combinations of boolean flags born of implementation errors.

Reply:
Ah crap this is breaking though. Maybe we can't change all of them yet but maybe for v3?